src/bridges/erc4626/VaultBridge.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright 2020 Spilsbury Holdings Ltd
+pragma solidity ^0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {AztecTypes} from "../../aztec/libraries/AztecTypes.sol";
+import {IERC4626} from "../../interfaces/erc4626/IERC4626.sol";
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+
+/**
+ * @title Aztec Connect Bridge for ERC4626 compatible vaults
+ * @author (@johhonn on Github)
+ * @notice You can use this contract to stake asset tokens and bridge share tokens for any ERC2626 compatible bridge.
+ * @dev Implementation of the IDefiBridge interface for IERC4626
+ *
+ 
+ */
+contract VaultBridge is BridgeBase {
+    using SafeERC20 for IERC20;
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {}
+
+    /**
+     * @notice Function which stakes or unstakes
+     * @dev This method can only be called from the RollupProcessor.sol. If the input asset is the ERC4626 vault asset, staking flow is
+     * executed. If input is the vault share token, unstaking. RollupProcessor.sol has to transfer the tokens to the bridge before calling
+     * the method. If this is not the case, the function will revert (either in STAKING_CONTRACT.stake(...) or during
+     * SB burn).This contract is stateless and cannot hold tokens thus is safe to approve.
+     *
+     * @param _inputAssetA - Vault asset (Staking) or Vault Address (Unstaking)
+     * @param _outputAssetA - Vault Address (Staking) or Vault asset (UnStaking)
+     * @param _totalInputValue - the amount of assets to stake or shares to unstake
+     * @return _outputValueA - the amount of shares (Staking) or assets (Unstaking) minted/transferred to the RollupProcessor.sol
+     *
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _totalInputValue,
+        uint256,
+        uint64,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 _outputValueA,
+            uint256,
+            bool
+        )
+    {
+        // // ### INITIALIZATION AND SANITY CHECKS
+
+        if (_isValidPair(_outputAssetA.erc20Address, _inputAssetA.erc20Address)) {
```

Comment:
Don't use `_isValidPair(...)` here and instead decide whether the user wants to enter or exit by passing 0 or 1 in the `_auxData`. The valid pair check is unnecessary because it will be done in the `listVault(...)` function. This will save quite a bit of gas.